hw/ip/aes/doc/_index.md

```diff
@@ -315,12 +326,111 @@ The processor writes both input data as well as the initial key to dedicated reg
+The provided implementation has a latency of 5 clock cycles per S-Box evaluation.
+As a result, the overall latency for processing a 16-byte data block increases from 12/14/16 to 56/66/72 clock cycles in AES-128/192/256 mode, respectively.
+Alternatively, the two original versions of the masked Canright S-Box can be chosen as proposed by [Canright and Batina: "A very compact "perfectly masked" S-Box for AES (corrected)".](https://eprint.iacr.org/2009/011.pdf)
+These are fully combinational (one S-Box evaluation every cycle) and have lower area footprint, but they are significantly less resistant to SCA.
+They are mainly included for reference but their usage is discouraged.
+
+Note that the masking can be enabled/disabled via compile-time Verilog parameter.
+It may be acceptable to disable the masking when using the AES cipher core for random number generation e.g. inside [CSRNG.]({{< relref "hw/ip/csrng/doc" >}})
+When disabling the masking, also an unmasked S-Box implementation needs to be selected using the corresponding compile-time Verilog parameter.
+When disabling masking, it is recommended to use the unmasked Canright or LUT S-Box implementation for ASIC or FPGA targets, respectively.
+Both are fully combinational and allow for one S-Box evaluation every clock cycle.
+
+### Fully-Parallel Data Path
+
+Any 1st-order masking scheme primarily protects against 1st-order SCA.
+Vulnerabilities against higher-order SCA might still be present.
+A common technique to aggravate higher-order attacks is to increase the noise in the system e.g. by leveraging parallel architectures.
+To this end, the AES cipher core uses a 128-bit parallel data path with a total of up to 20 S-Boxes (16 inside SubBytes, 4 inside KeyExpand) that are evaluated in parallel.
+
+Besides more noise for increased resistance against higher-order SCA, the fully-parallel architecture also enables for higher performance and flexibility.
+It allows users to seamlessly switch out the S-Box implementation in order to experiment with different masking schemes.
+To interface the data paths with the S-Boxes, a handshake protocol is used.
+
+### Note on Reset vs. Non-Reset Flip-Flops
+
+The choice of flip-flop type for registering sensitive assets such as keys can have implications on the vulnerability against e.g. combined reset glitch attacks and SCA.
+Following the [OpenTitan non-reset vs. reset flops rationale](https://github.com/lowRISC/opentitan/issues/2603), the following observations can be made:
+- If masking is enabled, key and state values are stored in two shares inside the AES unit.
+  Neither the Hamming weights of the individual shares nor the summed Hamming weight are proportional to the Hamming weight of the secret asset.
+- Input/output data and IV values are (currently) not stored in multiple shares but these are less critical as they are used only once.
```

Comment:
I'm quite sure you know this, and I don't think we want to get into the discussion in this documentation, but I'm only thinking of someone potentially extracting this IP and using it outside of OpenTitan context, so for completeness I'll say it:

In OpenTitan, the AES is not used to form the "identity ladder" - we use KMAC for this.  But, if one's application were to ladder AES such that data out of one AES operation became the key for the next stage in the ladder, then we would probably have to keep input/output data in 2 shares and not combine them - or have a way to load data out to key registers, at a minimum, without combining the shares.

Reply:
It wasn't fully clear to me why providing the output in two shares makes sense in previous discussions. But your concise explanation clarified things to me. Thanks for bringing this up! 

Actually, I think it wouldn't hurt to state this in the spec. I've added a little paragraph on this under "1st-order masking of the cipher core". But I can remove it if you think we better don't state that explicitly in the spec.